### PR TITLE
Voxel Light Source Position Customization

### PR DESCRIPTION
**Pull request by Metadorius**
TODO
- [x] transform into world coords
- [x] docs

Allows customizing the direction of voxel light in order to provide better visuals ingame.

In `rulesmd.ini`
```ini
[AudioVisual]
VoxelLightSource=0.02,-0.69,0.36
VoxelShadowLightSource=0.02,-0.69,0.36
```

The values are unset by default. The example values are according to the convo in Discord where @thomassneddon and CCrisp mentioned that those coords correspond are used to produce the following (apparently in radians? not sure)

> This is the value of projected light source direction i mentioned above, and as a result, the world space direction of YR VXL light is approx 0.20, -0.91, -0.36

On stock voxels, VPL, palette this looks like this, vanilla vs the above:

vanilla
tweaked

### CODE REVIEW

**Comment by Metadorius** [comment]
image

File: docs/Fixed-or-Improved-Logics.md [hunk 1]
```diff
@@ -770,6 +770,24 @@ ShadowIndex.Frame=0   ; integer (HVA animation frame index)
...
+![image](_static/images/VoxelLightSourceComparison.png)
+*New lighting with `VoxelLightSource=0.02,-0.69,0.36` vs default lighting, Prism Tank voxel by [CCS_qkl](https://bbs.ra2diy.com/home.php?mod=space&uid=20016&do=index)*
+
+- It is now possible to change the position of the light relative to the voxels. This allows for better lighting to be set up.
+  - Only the direction of the light is accounted, the distance to the voxel is not accounted.
+
+In `rulesmd.ini`:
+```ini
+[AudioVisual]
+VoxelLightSource= ; X,Y,Z - position of the light
```
Comment (Starkku) [inline docs/Fixed-or-Improved-Logics.md:784]: The value description probably needs to be something more descriptive than just this, at the very least it should mention that they're floating point values.

**Review by Starkku — CHANGES_REQUESTED** [review]
Looks good to me for most part.

File: docs/Fixed-or-Improved-Logics.md (hunk 1 above)
Comment (Metadorius) [inline docs/Fixed-or-Improved-Logics.md:784]:
```suggestion
VoxelLightSource=  ; X,Y,Z - position of the light, floating point values
```

**Review by Starkku — APPROVED** [review]

**Pull request merged**